Basic Scalafiddle integration

andreaTP opened the pull request:
Added a basic integration with ScalaFiddle.

andreaTP commented:
targeting #12 :-)
a bit of refactoring (probably need more at some point)

eed3si9n commented on core/src/main/scala/com/lightbend/paradox/markdown/Directive.scala:
```diff
@@ -1,12 +1,12 @@
 /*
  * Copyright © 2015 - 2016 Lightbend, Inc. <http://www.lightbend.com>
- * 
+ *
```
Was this whitespace manually edited? It should be managed by sbt-header.

eed3si9n commented:
Could you add tests please?

andreaTP commented:
fixed space in header and added some basic test...
please let me know if you need more or a minimum coverage or whatever, I'm not confident that everything is stable and things will not change soon, so I start slow ... 

b.t.w.
great tool! thanks!

pvlugter commented on core/src/main/scala/com/lightbend/paradox/markdown/Directive.scala:
```diff
@@ -135,6 +135,58 @@ object SnipDirective {
+      )
+    } catch {
+      case e: FileNotFoundException =>
+        throw new SnipDirective.LinkException(s"Unknown fiddle [${e.getMessage}] referenced from [${page.path}]")
```
Looks like this should be `FiddleDirective.LinkException`.

andreaTP replied:
true! thanks

pvlugter commented:
Cool. Nice feature.

andreaTP commented:
fix and rebased

andreaTP commented:
the failed test isn't related to this feature:
```
[info] - should parse and render all header children *** FAILED ***
[info]   "...h
[info]   <code>code</code>&[amp;]hellip;</h1>
[info]   
[info]   " did not equal "...h
[info]   <code>code</code>&[]hellip;</h1>
[info]   
[info]   " (AnchorLinkSpec.scala:29)
```

andreaTP commented:
Anything yet I should do here?

eed3si9n commented:
Restarted the Travis job.

eed3si9n commented:
Merging because testing locally succeeded, but I'm guessing there's some race condition on the tests that we need to take care of eventually.

pvlugter commented:
Haven't seen test failures like this before either, locally or in CI. In the past it was running on Lightbend/Typesafe Jenkins builds, not Travis. Looks like it's always one of the http render checks, but not the same one?

andreaTP commented:
thanks guys!
